src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -769,7 +769,15 @@ void shouldAuthenticate(boolean shouldAuthenticate) {
      */
     public static boolean isResponseRetryable(int responseCode, BoxAPIException apiException) {
         String message = apiException.getMessage();
-        Boolean isClockSkewError =  responseCode == 400 && message.contains("exp");
+        String errorCode = "";
+        try {
+            JsonObject responseBody = JsonObject.readFrom(apiException.getResponse());
+            errorCode = responseBody.get("code").toString();
+        } catch (Exception e) { }
```

Comment:
I believe in Java, it's better to check for existence of "code" than to rely on the overhead of throwing an exception, if it doesn't exist.

Reply:
Do you still need the `try … catch` now?

Reply:
What if the readFrom fails, such as when the body is null on a 500 error?

Reply:
We should check for that case, as well, then, for the same reason as above.

Reply:
Have to put the try catch back again because of a test case that wasn't passing [here](https://github.com/box/box-java-sdk/blob/69b2d57239ff41abcab895f5e0d30247ec39361b/src/test/java/com/box/sdk/BoxAPIResponseExceptionTest.java#L152). Apparently, an html body can be returned. In this case, I don't know how to do a check that the body is not HTML other than the try catch.